crates/oxc_linter/src/rules/react/no_unknown_property.rs

```diff
@@ -0,0 +1,734 @@
+use itertools::Itertools;
+use lazy_static::lazy_static;
+use once_cell::sync::Lazy;
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeName},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use phf::{phf_map, phf_set, Map, Set};
+use regex::Regex;
+use serde::Deserialize;
+use std::collections::hash_map::HashMap;
+use std::collections::hash_set::HashSet;
+
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{get_element_type, get_prop_name},
+    AstNode,
+};
+
+#[derive(Debug, Error, Diagnostic)]
+enum NoUnknownPropertyDiagnostic {
+    #[error("eslint-plugin-react(no-unknown-property): Invalid property found")]
+    #[diagnostic(severity(warning), help("Property '{1}' is only allowed on: {2}"))]
+    InvalidPropOnTag(#[label] Span, String, String),
+    #[error("eslint-plugin-react(no-unknown-property): React does not recognize data-* props with uppercase characters on a DOM element")]
+    #[diagnostic(severity(warning), help("Use '{1}' instead"))]
+    DataLowercaseRequired(#[label] Span, String),
+    #[error("eslint-plugin-react(no-unknown-property): Unknown property found")]
+    #[diagnostic(severity(warning), help("Use '{1}' instead"))]
+    UnknownPropWithStandardName(#[label] Span, String),
+    #[error("eslint-plugin-react(no-unknown-property): Unknown property found")]
+    #[diagnostic(severity(warning), help("Remove unknown property"))]
+    UnknownProp(#[label] Span),
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUnknownProperty(Box<NoUnknownPropertyConfig>);
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NoUnknownPropertyConfig {
+    #[serde(default)]
+    ignore: HashSet<String>,
+    #[serde(default)]
+    require_data_lowercase: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow usage of unknown DOM property.
+    ///
+    /// ### Why is this bad?
+    /// You can use unknown property name that has no effect.
+    ///
+    /// ### Example
+    /// ```jsx
+    ///  // Unknown properties
+    ///  const Hello = <div class="hello">Hello World</div>;
+    ///  const Alphabet = <div abc="something">Alphabet</div>;
+    ///  
+    ///  // Invalid aria-* attribute
+    ///  const IconButton = <div aria-foo="bar" />;
+    /// ```
+    NoUnknownProperty,
+    correctness
+);
+
+lazy_static! {
+    static ref ATTRIBUTE_TAGS_MAP: Map<&'static str, Set<&'static str>> = phf_map! {
```

Comment:
since all of the `static ref...` are compile time. i don't think we ned the `lazy_static!`

Reply:
https://docs.rs/phf/0.11.2/phf/#example-with-the-macros-feature-enabled

Reply:
Should it be `Lazy` or just `const`? If it will not be lazy we will allocate more memory eagerly, isn't it?

Reply:
just const i think

Reply:
I may not be 100% correct here. But since phf_maps are compile time there wrapping it in a lazy won't really have an effect